### ISSUE TITLE
[Hold for payment 01-02-2023] [$1000] Tooltip is not in the center while hovering on a profile pic

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#13940]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open any chat
2. hover mouse on profile pic

## Expected Result: 
tooltip should be in center of profile pic

## Actual Result:  
tooltip is not in the center

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] Web / Chrome / Safari 

**Version Number:** 1.2.47-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
![Screenshot 2022-12-28 at 3 22 25 PM](https://user-images.githubusercontent.com/43996225/210277675-62b4e748-5807-48d3-b7d4-34067799860f.png)
<img width="960" alt="Untitled" src="https://user-images.githubusercontent.com/43996225/210277684-6ae8aa98-b99f-426f-aabd-576b7b4d1e46.png">

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1672221180234109

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e3f915ebae1ec606</li>
        <li>Upwork Job ID: 1610419163878563840</li>
        <li>Last Price Increase: 2023-01-03</li>
    </ul>
</details>

**Comment by trjExpensify** [comment]
I can repro this reliably, moving `external`.

**Comment by Pujan92** [comment]
**Proposal**

Issue is occuring due to the margin applies to the child component of the Tooltip. 
To fix we need to apply margin right to the Pressable(parent component of tooltip).

```diff
diff --git a/src/pages/home/report/ReportActionItemSingle.js b/src/pages/home/report/ReportActionItemSingle.js
index a4958ab1b..4ad15b159 100644
--- a/src/pages/home/report/ReportActionItemSingle.js
+++ b/src/pages/home/report/ReportActionItemSingle.js
@@ -66,7 +66,7 @@ const ReportActionItemSingle = (props) => {
     return (
         <View style={props.wrapperStyles}>
             <Pressable
-                style={styles.alignSelfStart}
+                style={[styles.alignSelfStart, styles.mr2]}
                 onPressIn={ControlSelection.block}
                 onPressOut={ControlSelection.unblock}
                 onPress={() => showUserDetails(props.action.actorEmail)}
diff --git a/src/styles/styles.js b/src/styles/styles.js
index b3de12803..f4d161c59 100644
--- a/src/styles/styles.js
+++ b/src/styles/styles.js
@@ -695,7 +695,6 @@ const styles = {
     // Actions
     actionAvatar: {
         borderRadius: 20,
-        marginRight: 8,
     },
```

We can replace `borderRadius: 20` with the variable too.
`borderRadius: variables.componentBorderRadiusRounded`

**Comment by bondydaa** [comment]
This seems okay though I think it does end up making the profile pictures ever so slightly smaller as well

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/4073354/210460501-035268fd-eaa9-4dcd-b82d-6685376e16ab.png">

cc @shawnborton thoughts?

**Comment by bondydaa** [comment]
oh wait no nevermind it's not smaller, zooming in on that and using something for guide they're the same size 
<img width="227" alt="image" src="https://user-images.githubusercontent.com/4073354/210461119-1c2ca3d2-3594-48ca-b03a-322dc9c4a8f0.png">

👍

**Comment by bondydaa** [comment]
Will wait for the C+ to confirm as well

**Comment by syedsaroshfarrukhdot** [comment]
Proposal :-

We could simply fix this issue by setting `marginLeft : 8` 

 ``` diff
diff --git a/src/styles/styles.js b/src/styles/styles.js
index 82e9ca8f2..f86a8069d 100644
--- a/src/styles/styles.js
+++ b/src/styles/styles.js
@@ -696,6 +696,7 @@ const styles = {
     actionAvatar: {
         borderRadius: 20,
         marginRight: 8,
+        marginLeft: 8,
     },
 
     componentHeightLarge: {
     
  ```

**Comment by Pujan92** [comment]
@bondydaa Currently that avatar contains the right margin which increases the width of the Tooltip area and due to that center of the area is a bit right and not within the center of the avatar.

<img width="440" alt="Screenshot 2023-01-04 at 9 45 09 AM" src="https://user-images.githubusercontent.com/14358475/210483950-9486057e-6e0f-4dd4-a8b9-2e00339870a1.png">

**Comment by shawnborton** [comment]
I'm not sure at what point this changed but there should be 12px margin (not 8px) in between the avatar and the name/message. We could fix that here or we could save that for a new issue.

**Comment by syedsaroshfarrukhdot** [comment]
Proposal :- 


as according to recent point we need 12px margin in avatar and name
``` diff
diff --git a/src/pages/home/report/ReportActionItemSingle.js b/src/pages/home/report/ReportActionItemSingle.js
index a4958ab1b..0d5f4bfd4 100644
--- a/src/pages/home/report/ReportActionItemSingle.js
+++ b/src/pages/home/report/ReportActionItemSingle.js
@@ -65,19 +65,19 @@ const ReportActionItemSingle = (props) => {
 
     return (
         <View style={props.wrapperStyles}>
+             <Tooltip text={props.action.actorEmail}>
             <Pressable
-                style={styles.alignSelfStart}
+                style={[styles.alignSelfStart,styles.m3]}
                 onPressIn={ControlSelection.block}
                 onPressOut={ControlSelection.unblock}
                 onPress={() => showUserDetails(props.action.actorEmail)}
             >
-                <Tooltip text={props.action.actorEmail}>
-                    <Avatar
-                        containerStyles={[styles.actionAvatar]}
-                        source={avatarUrl}
-                    />
-                </Tooltip>
+                <Avatar
+                    containerStyles={[styles.actionAvatar]}
+                    source={avatarUrl}
+                /> 
             </Pressable>
+            </Tooltip>
             <View style={[styles.chatItemRight]}>
                 {props.showHeader ? (
                     <View style={[styles.chatItemMessageHeader]}>
diff --git a/src/styles/styles.js b/src/styles/styles.js
index 82e9ca8f2..04028f1c3 100644
--- a/src/styles/styles.js
+++ b/src/styles/styles.js
@@ -695,7 +695,6 @@ const styles = {
     // Actions
     actionAvatar: {
         borderRadius: 20,
-        marginRight: 8,
     },
 
     componentHeightLarge: {
diff --git a/src/styles/utilities/spacing.js b/src/styles/utilities/spacing.js
index 6e5de8b52..d4e0540a1 100644
--- a/src/styles/utilities/spacing.js
+++ b/src/styles/utilities/spacing.js
@@ -12,6 +12,9 @@ export default {
     m2: {
         margin: 8,
     },
+    m3: {
+        margin: 12,
+    },
 
     m4: {
         margin: 16,
```

**Comment by shawnborton** [comment]
Actually sorry, let's tackle the extra margin as a separate issue - I think there are other implications here we need to consider beyond the tooltip so I don't want to bloat the original issue at hand. Sorry about that!

**Comment by Pujan92** [comment]
Yes, 12px for particular this case will shift the content more to the right which will be inconsistent with the other messages alignment and requires a fix at other places too.

**Comment by Puneet-here** [comment]
### Proposal

This issue is present for name tooltip too.
<img width="181" alt="Screenshot 2023-01-04 at 4 03 56 PM" src="https://user-images.githubusercontent.com/101883770/210537468-5ec93a1c-3307-4b60-ac8d-d573e4ec00e1.png">

to fix it we need to remove the padding from below 
https://github.com/Expensify/App/blob/034a6f0fcc142b81802a2c21114392833f050671/src/styles/styles.js#L1315-L1316

Add the padding styles below 
https://github.com/Expensify/App/blob/034a6f0fcc142b81802a2c21114392833f050671/src/pages/home/report/ReportActionItemSingle.js#L85

To fix the avatar tooltip issue we need to add styles.mr2 below
https://github.com/Expensify/App/blob/034a6f0fcc142b81802a2c21114392833f050671/src/pages/home/report/ReportActionItemSingle.js#L69
and remove margin here https://github.com/Expensify/App/blob/034a6f0fcc142b81802a2c21114392833f050671/src/styles/styles.js#L698

**Comment by Pujan92** [comment]
@shawnborton just to correct the current margin in the LHN for avatar and name is 16px. 

<img width="1286" alt="Screenshot 2023-01-04 at 4 20 34 PM" src="https://user-images.githubusercontent.com/14358475/210540316-59d97c58-4a36-4ff2-ad2c-9fdee43ed17e.png">


To make the same 16px margin for the messages we can apply below changes.


<img width="859" alt="Screenshot 2023-01-04 at 4 25 10 PM" src="https://user-images.githubusercontent.com/14358475/210540531-7eaa28a8-1f32-446c-9f5e-b77b60306829.png">


```diff
diff --git a/src/pages/home/report/ReportActionItemSingle.js b/src/pages/home/report/ReportActionItemSingle.js
index a4958ab1b..fdb1464be 100644
--- a/src/pages/home/report/ReportActionItemSingle.js
+++ b/src/pages/home/report/ReportActionItemSingle.js
@@ -66,7 +66,7 @@ const ReportActionItemSingle = (props) => {
     return (
         <View style={props.wrapperStyles}>
             <Pressable
-                style={styles.alignSelfStart}
+                style={[styles.alignSelfStart, styles.mr4]}
                 onPressIn={ControlSelection.block}
                 onPressOut={ControlSelection.unblock}
                 onPress={() => showUserDetails(props.action.actorEmail)}
diff --git a/src/styles/styles.js b/src/styles/styles.js
index 82e9ca8f2..4005ff276 100644
--- a/src/styles/styles.js
+++ b/src/styles/styles.js
@@ -695,7 +695,6 @@ const styles = {
     // Actions
     actionAvatar: {
         borderRadius: 20,
-        marginRight: 8,
     },
 
     componentHeightLarge: {
@@ -1289,7 +1288,7 @@ const styles = {
         flexShrink: 1,
         flexBasis: 0,
         position: 'relative',
-        marginLeft: 48,
+        marginLeft: 56,
     },
```

**Comment by Pujan92** [comment]
> @shawnborton just to correct the current margin in the LHN for avatar and name is 16px.

But to make it consistent we need to apply changes to the display user time line and in the composer input too, which seems to be the reason to me why it is 8px. 

<img width="560" alt="Screenshot 2023-01-04 at 4 53 49 PM" src="https://user-images.githubusercontent.com/14358475/210545061-4d52fbc5-3ff8-47dd-85af-cf6bd366983f.png">

**Comment by umair-upwork** [comment]
### **Proposal**

### **Problem**

This issue is occurring because of the margin added to the `<Avatar>` with `styles.actionAvatar`, which has a `marginRight` of 8. The `<Avatar>`is inside of the `<Tooltip>`, so adding a `marginRight` to  `<Avatar>` also adds some margin to  `<Tooltip>`, causing it to shift to the right instead of remaining centered.

https://github.com/Expensify/App/blob/034a6f0fcc142b81802a2c21114392833f050671/src/pages/home/report/ReportActionItemSingle.js#L75-L76


### **Solution**

We can remove `styles.actionAvatar` from `<Avatar>` in https://github.com/Expensify/App/blob/034a6f0fcc142b81802a2c21114392833f050671/src/pages/home/report/ReportActionItemSingle.js#L75-L76

and add it to `<Pressable style={[styles.alignSelfStart, styles.actionAvatar]}>` in
https://github.com/Expensify/App/blob/034a6f0fcc142b81802a2c21114392833f050671/src/pages/home/report/ReportActionItemSingle.js#L68-L69

Now, if we add any `margin` value to it in
https://github.com/Expensify/App/blob/034a6f0fcc142b81802a2c21114392833f050671/src/styles/styles.js#L696-L699
### Tooltip will always remain centered while hovering on a profile pic

### **Demo**
https://user-images.githubusercontent.com/79830748/210546508-268de712-185a-4c8f-a0e6-7f040ba233ae.mov

**Comment by syedsaroshfarrukhdot** [comment]
Proposal :- 

The same issue is being replicated for name as highlighted by @Puneet-here 

As there are multiple ways to solve it I find this solution to be best working if we in future we need to change paddings for either item for name and avatar `Tooltip` position won't be affected it will be always be centered for both name and avatar.

We can pass avatar style to presssable and paddingLeft : 5 To ReportActionItemDate

``` diff
diff --git a/src/pages/home/report/ReportActionItemDate.js b/src/pages/home/report/ReportActionItemDate.js
index 0cf570132..02e928cf9 100644
--- a/src/pages/home/report/ReportActionItemDate.js
+++ b/src/pages/home/report/ReportActionItemDate.js
@@ -13,7 +13,7 @@ const propTypes = {
 };
 
 const ReportActionItemDate = props => (
-    <Text style={[styles.chatItemMessageHeaderTimestamp]}>
+    <Text style={[styles.chatItemMessageHeaderTimestamp, {...props.style}]}>
         {props.datetimeToCalendarTime(props.created)}
     </Text>
 );
diff --git a/src/pages/home/report/ReportActionItemSingle.js b/src/pages/home/report/ReportActionItemSingle.js
index a4958ab1b..980208df2 100644
--- a/src/pages/home/report/ReportActionItemSingle.js
+++ b/src/pages/home/report/ReportActionItemSingle.js
@@ -66,14 +66,13 @@ const ReportActionItemSingle = (props) => {
     return (
         <View style={props.wrapperStyles}>
             <Pressable
-                style={styles.alignSelfStart}
+                style={[styles.alignSelfStart, styles.actionAvatar]}
                 onPressIn={ControlSelection.block}
                 onPressOut={ControlSelection.unblock}
                 onPress={() => showUserDetails(props.action.actorEmail)}
             >
                 <Tooltip text={props.action.actorEmail}>
                     <Avatar
-                        containerStyles={[styles.actionAvatar]}
                         source={avatarUrl}
                     />
                 </Tooltip>
@@ -98,7 +97,7 @@ const ReportActionItemSingle = (props) => {
                                 />
                             ))}
                         </Pressable>
-                        <ReportActionItemDate created={props.action.created} />
+                        <ReportActionItemDate created={props.action.created} style={{paddingLeft : 5}}/>
                     </View>
                 ) : null}
                 {props.children}
diff --git a/src/styles/styles.js b/src/styles/styles.js
index 82e9ca8f2..be7715143 100644
--- a/src/styles/styles.js
+++ b/src/styles/styles.js
@@ -1312,7 +1312,6 @@ const styles = {
         fontSize: variables.fontSizeNormal,
         fontWeight: fontWeightBold,
         lineHeight: variables.lineHeightXLarge,
-        paddingRight: 5,
         paddingBottom: 4,
         ...wordBreak.breakWord,
     },
 ```
 
After Fix :-

https://user-images.githubusercontent.com/81307212/210563128-97293e69-27ae-44f7-a0e4-be42586aee70.mov

**Comment by Pujan92** [comment]
Just to make a point, the items mentioned by the other contributors are suggestions and improvements which is not directly related to the raised issue. I definitely agree that these are the bugs that can be raised in the future and solving them now is a good step but I believe the first valid proposal for the raised issue needs to be considered and should get a chance to incorporate these changes(all tooltip positional) if we are planning for it.

**Comment by shawnborton** [comment]
I agree that the download button probably doesn't need the extra margin.

**Comment by eVoloshchak** [comment]
@Pujan92's [proposal](https://github.com/Expensify/App/issues/13940#issuecomment-1370330887) looks good, I like the simple solution of applying styles to the correct component 

As @Puneet-here has [pointed out](https://github.com/Expensify/App/issues/13940#issuecomment-1370769241), the same problem occurs for the name tooltip. Should we fix it as part of this issue? Is there a more universal approach that would ensure we don't have the same problem on other pages?

**Comment by Pujan92** [comment]
> Is there a more universal approach that would ensure we don't have the same problem on other pages?

@eVoloshchak as per the current implementation I am not seeing any universal approach and we need to check & get rid of the margin/padding applied in the `Tooltip` children elements to fix the positional issues. But in the future implementation, we can avoid giving margin/padding to the `Tooltip` children.

**Comment by bondydaa** [comment]
Posted on this thread to get things cleared up https://expensify.slack.com/archives/C049HHMV9SM/p1672958081633489?thread_ts=1672817918.639959&cid=C049HHMV9SM

Will update here once there is agreement.

**Comment by bondydaa** [comment]
Okay cool got some agreement in that thread @Pujan92 we're accepting your proposal for this GH and will spin up a new GH for @daraksha-dk to handle the changes for `Header buttons` on https://github.com/Expensify/App/blob/034a6f0fcc142b81802a2c21114392833f050671/src/components/HeaderWithCloseButton.js#L141-L155 from https://github.com/Expensify/App/issues/13940#issuecomment-1371332290

**Comment by Pujan92** [comment]
PR is ready for review.

**Comment by Puneet-here** [comment]
Hi, I wanted to confirm if I am eligible for compensation since I mentioned a new instance of this issue and proposed a solution which was used. Like [here](https://github.com/Expensify/App/issues/13343#issuecomment-1378256046).

**Comment by eVoloshchak** [comment]
Not overdue, PR was deployed to production 3 days ago

**Comment by jatinsonijs** [comment]
@Pujan92 this issue still exist

<img width="1342" alt="Screenshot 2023-01-19 at 5 57 11 PM" src="https://user-images.githubusercontent.com/114683042/213443040-c938f711-dde1-4eae-abbd-b96993bea309.png">

**Comment by Pujan92** [comment]
Ohh, seems this [PR](https://github.com/Expensify/App/pull/13645/files) from @grgia overwritten changes which I did for this fix.

**Comment by bondydaa** [comment]
oh hm dang I do not know what we're supposed to do here, going to ask in slack.

**Comment by bondydaa** [comment]
to recap the slack discussion we're just going to have @Pujan92 re-add the fixes into the PR https://github.com/Expensify/App/pull/14247 to get it re-applied.

Georgia has been OOO and since normally we'd get the PR author who "broke" it to fix it let's bypass that this one time to get the fixes back in place and keep moving forward on these tooltip problems.

**Comment by bondydaa** [comment]
slack thread was here for posterity https://expensify.slack.com/archives/C01GTK53T8Q/p1674149953675229

**Comment by jatinsonijs** [comment]
@bondydaa just want to confirm do it will consider as regression report ?

https://github.com/Expensify/App/issues/13940#issuecomment-1396904118

**Comment by bondydaa** [comment]
yeah I think so 👍

**Comment by bondydaa** [comment]
merged https://github.com/Expensify/App/pull/14247 which is taking care of fixing the regression that got added.

**Comment by trjExpensify** [comment]
This hit prod 7 days ago (Wednesday, 1st Feb). Title didn't update, but doing that now to track it better. Is this the correct summary of payments due? 

$250 for @gadhiyamanan for the bug report
$250 for @jatinsonijs for the bug report after someone else's PR broke this again 
$1,500 for @Pujan92 for the fix + 50% bonus for PR merged within 3 days
$1,500 to @eVoloshchak for the C+ review + 50% bonus for PR merged within 3 days (we aren't counting someone's regression against this PR)

Let me know and I can ship the offers. Thanks!

**Comment by Puneet-here** [comment]
Hi @trjExpensify, I wanted to confirm if I am eligible for compensation since I mentioned a new instance of this issue and proposed a solution which was used. A similar case was [here](https://github.com/Expensify/App/issues/13343#issuecomment-1378256046).
![Screenshot_2023_0201_111434](https://user-images.githubusercontent.com/101883770/215961589-cb53734a-af40-49e7-abb0-4a94a4c3cf86.png)

**Comment by Pujan92** [comment]
@Puneet-here Just sharing my similar case [here](https://github.com/Expensify/App/issues/12957#issuecomment-1337390660) where it has counted as a suggestion. 😅

**Comment by trjExpensify** [comment]
Hey @Puneet-here, can you be more specific/direct with what your implying, please? I don't know what you're pointing to there with @JmillsExpensify's summary of payments on another issue? 😕 

The way I see it, this issue was created from a bug reported by @gadhiyamanan in #expensify-bugs as per the OP. We don't compensate for suggestions, and the solution we accepted was @Pujan92's.

**Comment by Puneet-here** [comment]
Sorry for not being specific earlier. I was sharing a similar case where two contributors were compensated because the code that was use in the final PR from both contributors (me and allrounderexperts) 


daraksha-dk mentioned one new instance of this issue and provided a solution, in their case a new issue was created and the issue was assigned to them for solving. 

In this issue I had added a new instance of this issue and provided the solution to fix that. The solution was used in the final PR. That's why I wanted to confirm if I am eligible for compensation since it is similar to above cases.

**Comment by trjExpensify** [comment]
Thanks! @daraksha-dk's issue was for the back/header buttons, which was treated separately because it had a different implementation as per [the outcome of this thread](https://expensify.slack.com/archives/C049HHMV9SM/p1672958081633489?thread_ts=1672817918.639959&cid=C049HHMV9SM). 

Your proposal on this issue wasn't accepted unfortunately, and the initial report came from @gadhiyamanan, so I don't think we treat it as an additional bug report - like we are the one where @jatinsonijs found a regression from another PR that reverted these changes.

**Comment by trjExpensify** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@Pujan92 / @eVoloshchak] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/4661
- [ ] [@Pujan92 / @eVoloshchak] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/4661#issuecomment-1400683633
- [ ]  [@Pujan92 / @eVoloshchak] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: Not sure if this is necessary, this should have been caught during the initial review, it's a visual bug
- [x]  [@trjExpensify] A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. - we aren't adding a regression test to the run for a purely visual bug

**Comment by Puneet-here** [comment]
By new "instance" I meant I reported and provided solution for a new place where this bug was present (it was also present in names, here's my [comment ](https://github.com/Expensify/App/issues/13940#issuecomment-1370769241) ) that instance was not part of the original issue but we decided to fix it here and since pujan had proposed the solution first for the original issue their proposal was taken and we included the fix for the new instance in the same PR so I think my case is similar as darkasha-dk 's, but in my case a new issue wasn't created.

**Comment by trjExpensify** [comment]
Ahh, I see. Thanks for elaborating on that, so yes, I'll ship you a bug reporting bonus.

**Comment by trjExpensify** [comment]
All done here!
